components/os/component_output.go

```diff
@@ -278,15 +279,19 @@ func getDefaultPoller() query.Poller {
 
 var getSystemdDetectVirtFunc = pkg_host.SystemdDetectVirt
 
-func createGet(cfg Config, eventBucket eventstore.Bucket) func(ctx context.Context) (_ any, e error) {
+func createGet(eventBucket eventstore.Bucket) func(ctx context.Context) (_ any, e error) {
 	return func(ctx context.Context) (_ any, e error) {
 		o := &Output{}
 
 		cctx, ccancel := context.WithTimeout(ctx, 15*time.Second)
 		virtEnv, err := getSystemdDetectVirtFunc(cctx)
 		ccancel()
 		if err != nil {
-			return nil, fmt.Errorf("failed to get virtualization environment using 'systemd-detect-virt': %w", err)
+			// ignore "context.DeadlineExceeded" since it's not a critical error and it's non-actionable
+			if !errors.Is(err, context.DeadlineExceeded) {
+				return nil, fmt.Errorf("failed to get virtualization environment using 'systemd-detect-virt': %w", err)
+			}
+			log.Logger.Warnw("failed to get virtualization environment using 'systemd-detect-virt'", "error", err)
```

Comment:
should we return here if failed to getSystemdDetectVirtFunc?  BTW, We just need to check if virtual env in initialization stage...

Reply:
we can improve it in next version